src/move_basic.cpp

```diff
@@ -688,6 +693,12 @@ bool MoveBasic::moveLinear(tf2::Transform& goalInDriving,
     tf2::Vector3 remaining = goalInBase.getOrigin();
     bool forward = (remaining.x() > 0);
 
+    // For abort check
+    double last = ros::Time::now().toSec();
+    double prevDistance = sqrt(remaining.x()*remaining.x() + remaining.y()*remaining.y());
+    double Pvelocity = 0;
```

Comment:
For naming consistency, should this be prevVelocity? I am not sure what this variable does.

Reply:
Renamed to controlVelocity. This velocity is calculated according to the distance to reference pose.